lib/Html/Parser.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+namespace OCA\Mail\Html;
+
+use DOMDocument;
+use function libxml_clear_errors;
+use function libxml_use_internal_errors;
+
+class Parser {
+
+	/**
+	 * Parse a DOM document from a string
+	 *
+	 * DOMDocument uses libxml to parse HTML and that expects HTML4. It
+	 * is likely that some markup cause an error, which is otherwise
+	 * logged. Therefore, we ignore any error here.
+	 * @todo Migrate to \Dom\HTMLDocument::createFromString when this app uses PHP8.4+
+	 */
+	public static function parseToDomDocument(string $html): DOMDocument {
+		$document = new DOMDocument();
```

Comment:
I think it should be "safe" to add something like $html === '' then return $document to avoid the psalm type coercion warning.

Reply:
I've pushed domdocument's type up to my little abstraction's parameter. Psalm is happy. Are you? :wink: